viz_scripts/energy_calculations.ipynb

```diff
@@ -215,28 +214,198 @@
+    "        print(\"\\n\")\n",
+    "        \n",
+    "        print(\"Total Kg CO2 Emissions\")\n",
+    "        print(ebco2['total_kg_CO2_emissions'])\n",
+    "     \n",
+    "    except:\n",
+    "        generate_missing_plot(plot_title_no_quality,debug_df,file_name)\n",
+    "        alt_text = store_alt_text_missing(debug_df, file_name, plot_title_no_quality)\n",
+    "# case: when label_options is absent from the config file\n",
+    "# else: \n",
+    "#     plot_title_no_quality=f\"Sketch of Total pounds of CO2 Emissions of {mode_of_interest} trips\"\n",
+    "#     file_name =f'sketch_CO2impact_{mode_of_interest}%s' % file_suffix\n",
+    "\n",
+    "#     try:\n",
+    "#         ebco2=data_eb.groupby('Replaced_mode').agg({'CO2_Impact(lb)': ['sum', 'mean']},)\n",
+    "#         ebco2.columns = ['total_lb_CO2_emissions', 'average_lb_CO2_emission']\n",
+    "#         ebco2 = ebco2.reset_index()\n",
+    "#         ebco2 = ebco2.sort_values(by=['total_lb_CO2_emissions'], ascending=False)\n",
+    "#         ebco2['boolean'] = ebco2['total_lb_CO2_emissions'] > 0\n",
+    "#         net_CO2_emissions = round(sum(ebco2['total_lb_CO2_emissions']), 2)\n",
+    "\n",
+    "#         x = ebco2['total_lb_CO2_emissions']\n",
+    "#         y = ebco2['Replaced_mode']\n",
+    "#         xLabel = 'CO2 Emissions (lb)'\n",
+    "#         yLabel = 'Replaced Mode'\n",
+    "#         color = ebco2['boolean']\n",
+    "\n",
+    "#         plot_title= plot_title_no_quality+f\"\\n Contribution by replaced mode towards a total of {net_CO2_emissions}(lb CO2 Emissions )\\n\"+quality_text\n",
+    "#         CO2_impact(x,y,color,plot_title,xLabel,yLabel,file_name)\n",
+    "#         alt_text = store_alt_text_bar(pd.DataFrame(x.values,y), file_name, plot_title)\n",
+    "\n",
+    "#         filtered_taxi_data = data_eb[data_eb['Replaced_mode'] == \"Taxi/Uber/Lyft\"]\n",
+    "#         filtered_bus_data = data_eb[data_eb['Replaced_mode'] == \"Bus\"]\n",
+    "#         filtered_notravel_data = data_eb[data_eb['Replaced_mode'] == \"No Travel\"]\n",
+    "#         filtered_freeshuttle_data = data_eb[data_eb['Replaced_mode'] == \"Free Shuttle\"]\n",
+    "#         filtered_walk_data = data_eb[data_eb['Replaced_mode'] == \"Walk\"]\n",
+    "#         filtered_GasCarShared_data = data_eb[data_eb['Replaced_mode'] == \"Gas Car, with others\"]\n",
+    "\n",
+    "#         selected_columns = ['distance', 'Replaced_mode_kg_CO2', 'Mode_confirm_kg_CO2', \"replaced_mode\", \"mode_confirm\"]\n",
+    "\n",
+    "#         print(\"With Default mapping:\")\n",
+    "#         print(\"\\n\")\n",
+    "\n",
+    "#         print(\"Walk Data:\")\n",
+    "#         print(str(filtered_walk_data[selected_columns].head()))\n",
+    "#         print(\"\\n\")\n",
+    "\n",
+    "#         print(\"No Travel Data:\")\n",
+    "#         print(str(filtered_notravel_data[selected_columns].head()))\n",
+    "#         print(\"\\n\")\n",
```

Comment:
why is this code duplicated?

Reply:
This is just a placeholder code, I wrote to compute and understand the CO2 emission. I intend to clean this up once I have validated my understanding about the difference in CO2 emission.

Reply:
See my note below